src/js/client/signup.js

```diff
@@ -0,0 +1,145 @@
+// provide a front-end to /api/user/signup
+/* global fluid, jQuery */
+(function ($) {
+    "use strict";
+    var gpii = fluid.registerNamespace("gpii");
+    fluid.registerNamespace("gpii.express.couchuser.frontend.signup");
+
+    // Try to log in and display the results
+    gpii.express.couchuser.frontend.signup.submit = function (that, event) {
+        if (event) { event.preventDefault(); }
+        var name     = that.locate("name").val();
+        var email    = that.locate("email").val();
+        var password = that.locate("password").val();
+        var confirm  = that.locate("confirm").val();
+
+        // Our user handling library doesn't offer password confirmation, so we have to do it ourselves for now
+        if (password !== confirm) {
+            that.displayError(null, null, "The passwords you have entered don't match.");
+            return;
+        }
+
+        // This is not at all nice, but in order to make jQuery correctly send array data, we have to:
+        //
+        // a) set `traditional` to `true`, otherwise the roles variable comes through with brackets appended to it.
+        // b) send at least a two-item array (a one-item array becomes a string variable)
+        //
+        // Thanks jQuery!
+        var settings = {
+            type:        "POST",
+            url:         that.options.apiUrl + "/signup",
+            success:     that.displayReceipt,
+            error:       that.displayError,
+            traditional: true,
```

Comment:
This took a look longer than I would like to figure out.  I made a standalone gist to demonstrate various methods of sending POST data:

https://gist.github.com/the-t-in-rtf/15852092712fd2cf211c

Form encoding corrupts the variable name.  This leaves us with two options:

1.  Continue sending form-encoded data with `traditional` set to `true`.
2. Add various options to force jQuery to send a JSON payload (see gist).

I plan to go through and use the second approach for all jQuery code which current has a `json:true` option.  It turns out I was confusing this with node's request options, the `json` option is meaningless for `jQuery.ajax()`, and thus all of these are defaulting to form encoding.